parser/parser_test.go

```diff
@@ -404,6 +404,17 @@ func TestWhereWithNotLike(t *testing.T) {
 
 }
 
+func TestDistinct(t *testing.T) {
+	New("select distinct author from commits")
+
+	_, err := AST()
+	if err != nil {
+		t.Errorf(err.Error())
+		return
+	}
+	// Maybe this test isn't really needed. It does increase coverage though.
```

Comment:
It looks useful but I would suggest to add a new check to make sure that the select.Distinct  is true.

Reply:
Good - it's been done :)